plugins/net_plugin/net_plugin.cpp

```diff
@@ -3598,7 +3598,8 @@ namespace eosio {
          hello.token = sha256();
       hello.p2p_address = my_impl->p2p_address;
       if( is_transactions_only_connection() ) hello.p2p_address += ":trx";
-      if( is_blocks_only_connection() ) hello.p2p_address += ":blk";
+      // if we are not accepting transactions tell peer we are blocks only
+      if( is_blocks_only_connection() || !my_impl->p2p_accept_transactions ) hello.p2p_address += ":blk";
```

Comment:
I think this might be better because it also updates `connection_type` and logs the change.

```c++
       if(hello.sig == chain::signature_type())
          hello.token = sha256();
       hello.p2p_address = my_impl->p2p_address;
+      if (!my_impl->p2p_accept_transactions && !is_blocks_only_connection()) {
+         fc_dlog( logger, "Setting connection ${c} type for: ${peer} to transactions only", ("c", connection_id)("peer", peer_addr) );
+         connection_type = blocks_only;
+      }
       if( is_transactions_only_connection() ) hello.p2p_address += ":trx";
       if( is_blocks_only_connection() ) hello.p2p_address += ":blk";
       hello.p2p_address += " - " + hello.node_id.str().substr(0,7);
```

Reply:
We do not want to set `connection_type = blocks_only` see my comment to Lin above.
Adding a log statement we are doing it is a good idea.

Reply:
Right now, we are keeping state describing what the connection permits in two places, `connection_type ` and `p2p_accept_transaction`.

I am suggesting keeping it only in one place: `connection_type `.

Reply:
That is not how I see it. `connection_type` is the type of connection I have with my peer. That does not imply it is the same connection they have with me. `p2p_accept_transaction` is an indication of what I allow. A peer might not honor my request for what they send me, so I need to enforce it myself. In my mind these are two separate concepts.

Reply:
Well, I understand this is how it is setup today. However, if you just consider what you allow on your connection to your peer:

- send_transactions
- accept_transactions
- send_blocks
- accept_blocks

I don't see why these capabilities should not be stored together in one field. It is certainly less confusing than the current way imho.

Reply:
However I'm fine approving as-is.